backend/src/BuiltinExecution/Libs/HttpClient.fs

```diff
@@ -24,15 +24,66 @@ type Request = { url : string; method : Method; headers : Headers.T; body : Body
 
 type Response = { statusCode : int; headers : Headers.T; body : Body }
 
-// forked from Elm's HttpError type
-// https://package.elm-lang.org/packages/elm/http/latest/Http#Error
-type RequestError =
-  | BadUrl of details : string
-  | Timeout
-  | NetworkError
-  | Other of details : string
 
-type RequestResult = Result<Response, RequestError>
+module HeaderError =
+  type HeaderError = | EmptyKey
+
+  let toDT (err : HeaderError) : Dval =
+    let (caseName, fields) =
+      match err with
+      | EmptyKey -> "EmptyKey", []
+    let typeName =
+      TypeName.fqPackage "Darklang" [ "Stdlib"; "HttpClient" ] "HeaderError" 0
+    DEnum(typeName, typeName, [], caseName, fields)
+
+module BadUrl =
+  type BadUrlDetails =
+    | UnsupportedProtocol
+    | InvalidHost
+    | InvalidUri
+    | InvalidRequest
+
+  let toDT (err : BadUrlDetails) : Dval =
+    let (caseName, fields) =
+      match err with
+      | UnsupportedProtocol -> "UnsupportedProtocol", []
+      | InvalidHost -> "InvalidHost", []
+      | InvalidUri -> "InvalidUri", []
+      | InvalidRequest -> "InvalidRequest", []
+
+    let typeName =
+      TypeName.fqPackage "Darklang" [ "Stdlib"; "HttpClient" ] "BadUrlDetails" 0
+    Dval.DEnum(typeName, typeName, [], caseName, fields)
+
+module RequestError =
+  // forked from Elm's HttpError type
+  // https://package.elm-lang.org/packages/elm/http/latest/Http#Error
+  type RequestError =
+    | BadUrl of BadUrl.BadUrlDetails
+    | Timeout
+    | HeaderError of HeaderError.HeaderError
+    | IOError
```

Comment:
I'm not seeing a real difference that we can point to for users between IOError and ConnectionError. Should we consolidate these into NetworkError (like Elm).